evap/evaluation/auth.py

```diff
@@ -45,160 +48,101 @@ def authenticate(self, request, email=None, password=None):  # pylint: disable=a
         return None
 
 
-def user_passes_test(test_func):
-    """
-    Decorator for views that checks whether a user passes a given test
-    (raising 403 if not). The test should be a callable that takes the
-    user object and returns True if the user passes.
-    """
-
-    def decorator(view_func):
-        @wraps(view_func)
-        def _wrapped_view(request, *args, **kwargs):
+def class_or_function_check_decorator(test_func: Callable[[UserProfile], bool]):
```

Comment:
Given that this is more complex now: I'd vote to add 4 unit tests here (function passed, function failed, class passed, class failed) if we don't already have something similar.